lib/gtfs_realtime_viz.ex

```diff
@@ -56,13 +57,75 @@ defmodule GTFSRealtimeViz do
   @spec visualize_diff(term, term, route_opts) :: String.t
   def visualize_diff(group_1, group_2, opts) do
     routes = Map.keys(opts)
-    archive_1 = get_vehicle_archive(group_1, routes)
-    archive_2 = get_vehicle_archive(group_2, routes)
-    [vehicle_archive: Enum.zip(archive_1, archive_2), routes: opts, render_diff?: true]
+    vehicle_archive_1 = get_vehicle_archive(group_1, routes)
+    trip_archive_1 = get_trip_update_archive(group_1, routes)
+    vehicle_archive_2 = get_vehicle_archive(group_2, routes)
+    trip_archive_2 = get_trip_update_archive(group_2, routes)
+    trip_archive = archive_trips(trip_archive_1, trip_archive_2)
+
+    [trip_update_archive: trip_archive, vehicle_archive: Enum.zip(vehicle_archive_1, vehicle_archive_2), routes: opts, render_diff?: true]
     |> gen_html()
     |> Phoenix.HTML.safe_to_string()
   end
 
+  defp archive_trips(trip_set_1, trip_set_2) do
+    archived = Enum.reduce(trip_set_1, %{}, fn {key, value}, acc ->
+      Map.put(acc, key, {value, trip_set_2[key]})
+    end)
+    archived = Enum.reduce(trip_set_2, archived, fn {key, value}, acc ->
+      Map.put(acc, key, {trip_set_1[key], value})
+    end)
+  end
+
+  defp get_trip_update_archive(group, routes) do
+    group
+    |> State.trip_updates
+    |> trips_we_care_about(routes)
+    |> trip_updates_by_stop_id
+  end
+
+  def trips_we_care_about(state, routes) do
+    Enum.map(state,
+      fn {descriptor, update_list} ->
+        filtered_positions = update_list
+        |> Enum.filter(fn trip_update ->
+          trip_update.trip && trip_update.trip.route_id in routes
```

Comment:
I was going to ask if we actually needed this function, since it seems identical to `vehicles_we_care_about`, but actually there is a small difference: the proto definition says that `trip_update.trip` is required to be present, so you don't need this `trip_update.trip &&` guard.